app/org/sagebionetworks/bridge/BridgeUtils.java

```diff
@@ -23,6 +31,28 @@ public static String generateGuid() {
         return UUID.randomUUID().toString();
     }
     
+    /**
+     * Generate a key that is based on the natural key of the task (schedule plan GUID, scheduled start time of the
+     * task, and the activity reference), with some hashing to reduce unusual characters such as URL path characters in
+     * the string. The same key will be generated each time for the same task, and the key should never collide with
+     * another key within the scope of an individual user's tasks (this is ensured by the fact that a schedule plan
+     * cannot contain a schedule with identical activities).
+     * 
+     * @param task
+     * @return
+     */
+    public static String generateTaskGuid(Task task) {
+        checkNotNull(task.getSchedulePlanGuid());
+        checkNotNull(task.getScheduledOn());
+        checkNotNull(task.getActivity());
+        checkNotNull(task.getActivity().getRef());
+        
+        HashFunction hf = Hashing.murmur3_128();
+        HashCode hc = hf.newHasher().putString(task.getActivity().getRef(), Charsets.UTF_8).hash();
+        return String.format("%s%s%s", task.getSchedulePlanGuid().replaceAll("-", ""), 
+            task.getScheduledOn(), hc.toString());
+    }
```

Comment:
Alternative: don't hash. Just put the string content into this identifier. Create a table mapping this identifier to a GUID, only return the GUID to a user.